src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -85,6 +96,13 @@ public async Task<IEnumerable<PluginDiscoveryResult>> DiscoverAsync(Cancellation
                 }
 
                 _pluginFiles = GetPluginFiles(cancellationToken);
+
+                if (string.IsNullOrEmpty(_rawPluginPaths))
+                {
+                    // Nuget plugin configuration environmental variables were not used to discover the configuration files
+                    _pluginFiles.AddRange(GetNetToolsPluginFiles());
```

Comment:
> Is rawPluginPaths the value of the NUGET_PLUGIN_PATHS or NUGET_NETFX_PLUGIN_PATHS environment variables when they're set?

Yes Andy, `rawPluginPaths` is the value of the NUGET_NETFX_PLUGIN_PATHS or NUGET_NETCORE_PLUGIN_PATHS or NUGET_PLUGIN_PATHS environment variables.

Reply:
perhaps `_overridePath` is clearer.

Plus XMLDoc comments that explains this, because even reading the docs, it's not super clear where the value for this `_rawPluginPaths` property comes from. Even knowing the override feature, it sounds to me like this value should never be null, and when the CLI arg is not provided, it should contain the default plugin path.

(also I'm lazy, so I don't want to read the docs, if the comments explain that the CLI arg allows the customer to override normal discovery, and discover only plugins in a specific directory, we won't need to context switch to the docs, and this is assuming the person reading the code even realises that there are relevant docs)

Reply:
I agree, I think the name _rawPluginPaths could be a bit misleading. How about _`envVariablePluginPaths`?